src/main/java/ay2122s1_cs2103t_w16_2/btbb/ui/tabcontent/StatTabContent.java

```diff
@@ -17,15 +26,43 @@
     @FXML
     private StackPane ingredientListPanelPlaceholder;
 
+    @FXML
+    private PieChart clientPieChart;
+
     /**
-     * Constructs a {@code StatTab}.
+     * Constructs a {@code StatTabContent}.
      *
      * @param ingredientList List of ingredients to display in the tab.
+     * @param topTenClients Top ten clients to be shown in statistics.
      */
-    public StatTabContent(ObservableList<Ingredient> ingredientList) {
+    public StatTabContent(ObservableList<Ingredient> ingredientList,
+                          List<Map.Entry<Pair<GenericString, Phone>, Integer>> topTenClients) {
         super(FXML);
 
         IngredientListPanel ingredientListPanel = new IngredientListPanel(ingredientList);
         ingredientListPanelPlaceholder.getChildren().add(ingredientListPanel.getRoot());
+
+        setClientStatisticsPieChart(topTenClients);
+    }
+
+    /**
+     * Converts a list of top ten clients to data points in the pie chart.
+     *
+     * @param topTenClients Top ten clients to be converted to data points in the pie chart.
+     */
+    private void setClientStatisticsPieChart(List<Map.Entry<Pair<GenericString, Phone>, Integer>> topTenClients) {
+        ObservableList<PieChart.Data> pieChartData = FXCollections.observableList(
+                topTenClients.stream()
+                        .map(entry -> {
+                            Pair<GenericString, Phone> clientNamePhonePair = entry.getKey();
```

Comment:
If we aren't displaying the count in the view, then why do we need to pass an entry to the view? We could have just passed the client & phone pair